packages/jsActions/nanoflow-actions-native/src/external/DownloadFile.ts

```diff
@@ -0,0 +1,45 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import fetchBlob from "rn-fetch-blob";
+
+// BEGIN EXTRA CODE
+function formatMendixFileUrl(file: mendix.lib.MxObject): string {
+    return `${mx.remoteUrl}file?guid=${file.getGuid()}&changedDate=${
+        file.get("changedDate") ?? ""
+    }&name=${encodeURIComponent(file.get("Name"))}`;
+}
+
+function formatPath(...pathArgs: string[]): string {
+    return pathArgs.filter(arg => !!arg).join("/");
+}
+// END EXTRA CODE
+
+/**
+ * @param {MxObject} file
+ * @param {string} filePath - optional
+ * @returns {Promise.<boolean>}
+ */
+export async function DownloadFile(file: mendix.lib.MxObject, filePath: string): Promise<boolean> {
+    // BEGIN USER CODE
+    if (!file) {
+        return Promise.reject(new Error("Input parameter 'file' is required"));
+    }
+    const dirs = fetchBlob.fs.dirs;
+    try {
+        await fetchBlob
+            .config({
+                path: formatPath(dirs.DownloadDir, filePath, <string>file.get("Name"))
```

Comment:
I just realized that we probably shouldn't blindly trust the `Name` attribute here either. I can imagine all sorts of fun if it starts with `../../../../`, for example ;). Some sanitization would be prudent.

Reply:
Sanitization is now in place :).